crates/nu-cli/src/commands/uniq.rs

```diff
@@ -30,17 +30,67 @@ impl WholeStreamCommand for Uniq {
     }
 }
 
-async fn uniq(args: CommandArgs, _registry: &CommandRegistry) -> Result<OutputStream, ShellError> {
+async fn uniq(args: CommandArgs, registry: &CommandRegistry) -> Result<OutputStream, ShellError> {
+    let args = args.evaluate_once(&registry).await?;
+    let should_show_count = args.has("count");
     let input = args.input;
-    let uniq_values: IndexSet<_> = input.collect().await;
+    let uniq_values = {
+        let mut counter = IndexMap::<nu_protocol::Value, usize>::new();
+        for line in input.into_vec().await {
+            *counter.entry(line).or_insert(0) += 1;
+        }
+        counter
+    };
 
     let mut values_vec_deque = VecDeque::new();
 
-    for item in uniq_values
-        .iter()
-        .map(|row| ReturnSuccess::value(row.clone()))
-    {
-        values_vec_deque.push_back(item);
+    if should_show_count {
+        for item in uniq_values {
+            use nu_protocol::{UntaggedValue, Value};
+            let value = {
+                match item.0.value {
+                    UntaggedValue::Row(mut row) => {
+                        row.entries.insert(
+                            "count".to_string(),
+                            UntaggedValue::int(item.1).into_untagged_value(),
+                        );
+                        Value {
+                            value: UntaggedValue::Row(row),
+                            tag: item.0.tag,
+                        }
+                    }
+                    UntaggedValue::Primitive(p) => {
+                        let mut map = IndexMap::<String, Value>::new();
+                        map.insert(
+                            "value".to_string(),
+                            UntaggedValue::Primitive(p).into_untagged_value(),
+                        );
+                        map.insert(
+                            "count".to_string(),
+                            UntaggedValue::int(item.1).into_untagged_value(),
+                        );
+                        Value {
+                            value: UntaggedValue::row(map),
+                            tag: item.0.tag,
+                        }
+                    }
+                    UntaggedValue::Table(_) => Value {
+                        value: UntaggedValue::Error(ShellError::labeled_error(
```

Comment:
This works.

A more idiomatic way, if you're interested, is to return the error directly, since the return type of the function lets you do that.

Reply:
Done.